### PR TITLE
Fix apparent typos in zstd:chunked tests

### PR DESCRIPTION
**Pull request by mtrmac**
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by Luap99 — COMMENTED** [review]
If this change did not cause the test to fail are we testing the right thing? Clearly this option does not seem to matter so why are we even setting it?

**Comment by giuseppe** [comment]
> If this change did not cause the test to fail are we testing the right thing? Clearly this option does not seem to matter so why are we even setting it?

we force to force the pull to the "partial pulls" code path even with traditional images.

We can improve the test to really check that it is what happens (and we could have caught the issue earlier doing so), but we need that flag to test the conversion code path.

**Comment by Luap99** [comment]
> We can improve the test to really check that it is what happens (and we could have caught the issue earlier doing so)

Well we must do that, without it the test seems like a NOP. If you want to test the "partial pulls" code path then the test MUST check that this code path is used. Passing the right cli option is not good enough. What if the option passing has a bug or we do not pass the option along the right code paths, etc...? If the test didn't fail before it will not fail when a regression causes the code path to not be executed.

**Comment by giuseppe** [comment]
@mtrmac could you please amend something like?

```diff
diff --git a/test/system/155-partial-pull.bats b/test/system/155-partial-pull.bats
index 961354bc7..e1673eb4a 100644
--- a/test/system/155-partial-pull.bats
+++ b/test/system/155-partial-pull.bats
@@ -84,9 +84,10 @@ EOF
 
     run_podman $globalargs rmi $image2 $image3
 
-    run_podman $globalargs pull \
+    run_podman --log-level debug $globalargs pull \
                $pushpullargs \
                $image2
+    assert "$output" =~ "Retrieved partial blob"
 
     run -0 skopeo inspect containers-storage:$image2
     assert "$output" =~ "application/vnd.oci.image.layer.v1.tar\+zstd" "pulled image must be zstd-compressed"
@@ -182,9 +183,10 @@ EOF
 
     run_podman $globalargs rmi $image
 
-    run_podman $globalargs pull \
+    run_podman --log-level debug $globalargs pull \
                $pushpullargs \
                $image
+    assert "$output" =~ "Retrieved partial blob"
 
     # expect that the image contains exactly the same data as before
     mount_image_and_take_digest $image
```

**Comment by mtrmac** [comment]
> we force to force the pull to the "partial pulls" code path even with traditional images.

That’s not what this option does, FWIW (that would be `convert_images`). This one enables partial pulls at all, they are now disabled by default.

**Comment by mtrmac** [comment]
Totally unrelated, but: https://cirrus-ci.com/task/6587238856785920 reports success, but the detailed logs https://api.cirrus-ci.com/v1/task/6587238856785920/logs/main.log show 4 failed tests (all in the setup phase). Is that expected???

@Luap99 any idea?

File: test/system/155-partial-pull.bats [hunk 1]
```diff
@@ -157,11 +158,12 @@ function mount_image_and_take_digest() {
 }
 
 @test "zstd chunked does not modify image content" {
+    skip_if_remote "need to mount an image"
```
Comment (Luap99) [inline test/system/155-partial-pull.bats:161]: the first skip is part of setup() which is a magic bats function that will be run for each test case so this here is redundant

**Review by Luap99 — COMMENTED** [review]
> Totally unrelated, but: https://cirrus-ci.com/task/6587238856785920 reports success, but the detailed logs https://api.cirrus-ci.com/v1/task/6587238856785920/logs/main.log show 4 failed tests (all in the setup phase). Is that expected???

Uhm no, like this is bad, very bad... I need to do some tests there if bats didn't exit > 0 (bats problem) or if our script that runs bats twice ignores the exit code somehow (unlikely in my eyes as I have be seen many test failures since we changed that)

File: test/system/155-partial-pull.bats (hunk 1 above)
Comment (mtrmac) [reply test/system/155-partial-pull.bats:161]: Yes… I was trying to capture “this is one more reason why the test won’t work” but I appreciate that having a redundant skip here might be confusing.

Comment (mtrmac) [reply test/system/155-partial-pull.bats]: Now with a comment.

File: test/system/155-partial-pull.bats [hunk 2]
```diff
@@ -11,7 +11,7 @@ load helpers.registry
 # BEGIN filtering - none of these tests will work with podman-remote
 
 function setup() {
-    skip_if_remote "none of these tests work with podman-remote"
+    skip_if_remote "zstd:chunked tests depend on start_registry, which does not work with podman-remote"
```
[the remark this answers is not in the document]
Comment (Luap99) [reply test/system/155-partial-pull.bats]: Yes I agree skips must have a meaningful reason. `skip_if_remote "doesn't work with remote"` is pointless as skip_if_remote said as much. It should list a proper reason why not, i.e. what you did although the actual reason why start_registry doe snot work is that it uses gloabal storage options.

Basically any test using global podman options will not work correctly in a remote context.

**Comment by Luap99** [comment]
> > > Totally unrelated, but: https://cirrus-ci.com/task/6587238856785920 reports success, but the detailed logs https://api.cirrus-ci.com/v1/task/6587238856785920/logs/main.log show 4 failed tests (all in the setup phase). Is that expected???
> > 
> > 
> > Uhm no, like this is bad, very bad...
> 
> I have split an attempt to reproduce that, + a bit of debug logging, into #24697 . I’d be perfectly happy to leave investigating this to an expert.

I will chase this down, don't worry about it. Running this through several CI cycles is pretty pointless for me I can easily run the test locally.

**Comment by mtrmac** [comment]
Updated:
- Added the sanity check that we are truly using the chunked pull code path, thanks to @giuseppe (and https://github.com/containers/podman/commit/466e1af0a6bd75381467e29ef12326a18cc4ac39 shows that this does cause failures without the original typo fix)
- … except on VFS, where we can’t be doing that (but keep the test running, to confirm that the fallback code path is not broken for chunked images)
- Tried to better document the reasons to skip the test
- Manually confirmed the remote tests are not failing now; a true fix will be #24698 .

Ready for review.

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**